src/components/tree/AposTree.vue

```diff
@@ -0,0 +1,154 @@
+          headers[header.name] = header.label;
+        });
+      }
+
+      let completeRows = [headers];
+      // Add child rows into `completeRows`.
+      this.data.rows.forEach(row => {
+        completeRows.push(row);
+
+        if (row.children && row.children.length > 0) {
+          this.nested = true;
+          completeRows = completeRows.concat(row.children);
+        }
+      });
+      completeRows = completeRows.slice(0, 50);
+
+      // Loop over the combined header/rows array, finding the largest value
+      // for each key.
+      completeRows.forEach(row => {
+        if (spacingRow.length === 0) {
+          spacingRow = Object.assign({}, row);
+          return;
+        }
+
+        this.data.headers.forEach(col => {
+          const key = col.name;
+          if (
+            !spacingRow[key] ||
+            (spacingRow[key] &&
+            spacingRow[key].toString().length < row[key].toString().length)
+          ) {
+            spacingRow[key] = row[key];
+          }
+        });
+      });
+      // Place that largest value on that key of the spacingRow object.
+      // Put that array in the DOM, and generate styles to be passed down based on its layout. Give the first column any leftover space.
+      const finalRow = [];
+      this.data.headers.forEach(col => {
+        let obj;
+        const foundIndex = this.data.headers.findIndex(o => {
+          return o.name === col.name;
+        });
+        const spacerInfo = {
+          name: col.name,
+          label: spacingRow[col.name]
+        };
+
+        if (foundIndex > -1) {
+          // Deep copy the original header column to capture all options.
+          const foundObj = JSON.parse(JSON.stringify(this.data.headers[foundIndex]));
+
+          if (foundObj.iconOnly) {
+            // If the "column" will only show icons, let the "column header"
+            // set the width.
+            delete spacerInfo.label;
+          }
+
+          obj = Object.assign(foundObj, spacerInfo);
+        } else {
+          obj = spacerInfo;
+        }
+
+        finalRow.push(obj);
+      });
+      return finalRow;
+    }
+  },
+  methods: {
+    setWidths(widths) {
+      this.colWidths = widths;
+    },
+    setBusy(val) {
+      this.$emit('busy', val);
+    },
+    update(event) {
+      this.$emit('update', {
+        // The ID of the item that moved.
+        changedId: event.item.dataset.rowId,
+        // The ID of the original parent, or 'root' if top-level.
+        startContext: event.from.dataset.listId,
+        // The index of the moved item within its original context.
+        startIndex: event.oldIndex,
+        // The ID of the new parent, or 'root' if top-level.
+        endContext: event.to.dataset.listId,
+        // The index of the moved item within its new context.
+        endIndex: event.newIndex
+      });
+      this.setBusy(false);
+    }
+  }
+};
+</script>
+
+<style lang="scss">
+@import './_tree';
```

Comment:
Should the styles just live here instead of being imported?

Reply:
The component was getting super large and hard to search within. I'll bring it back in.